gtsfm/loader/mobilebrick_loader.py

```diff
@@ -0,0 +1,135 @@
+            self._image_paths.append(image_path)
+
+        # Load GT intrinsics
+        self._intrinsics_dir = os.path.join(data_dir, "intrinsic")
+        self._intrinsics = []
+        for i in range(self._num_images):
+            intrinsics_file = os.path.join(self._intrinsics_dir, f"{i:06d}.txt")
+            K = np.loadtxt(intrinsics_file)
+            self._intrinsics.append(Cal3Bundler((K[0, 0] + K[1, 1]) / 2, 0, 0, K[0, 2], K[1, 2]))
+
+        # Load GT poses
+        self._poses_dir = os.path.join(data_dir, "pose")
+        self._wTi = []
+        for i in range(self._num_images):
+            pose_file = os.path.join(self._poses_dir, f"{i:06d}.txt")
+            wTi_mat = np.loadtxt(pose_file)
+            wTi = Pose3(Rot3(wTi_mat[:3, :3]), wTi_mat[:3, 3])
+            self._wTi.append(wTi)
+
+    def image_filenames(self) -> List[Path]:
+        """Return the file names corresponding to each image index."""
+        return [Path(fpath) for fpath in sorted(os.listdir(self._image_dir))]
+
+    def __len__(self) -> int:
+        """The number of images in the dataset.
+
+        Returns:
+            the number of images.
+        """
+        return self._num_images
+
+    def get_image_full_res(self, index: int) -> Image:
+        """Get the image at the given index, at full resolution.
+
+        Args:
+            index: the index to fetch.
+
+        Raises:
+            IndexError: if an out-of-bounds image index is requested.
+
+        Returns:
+            Image: the image at the query index.
+        """
+        if index < 0 or index >= len(self):
+            raise IndexError(f"Image index {index} is invalid")
+
+        # Read in image.
+        img = io_utils.load_image(self._image_paths[index])
+        return Image(value_array=img.value_array, exif_data=img.exif_data, file_name=img.file_name)
+
+    def get_camera_intrinsics_full_res(self, index: int) -> Cal3Bundler:
+        """Get the camera intrinsics at the given index, valid for a full-resolution image.
+
+        Args:
+            the index to fetch.
+
+        Returns:
+            intrinsics for the given camera.
+        """
+        if index < 0 or index >= len(self):
+            raise IndexError(f"Image index {index} is invalid")
+        intrinsics = self._intrinsics[index]
+
+        return intrinsics
+
+    def get_camera_pose(self, index: int) -> Optional[Pose3]:
+        """Get the camera pose (in world coordinates) at the given index.
+
+        Args:
+            index: the index to fetch.
+
+        Returns:
+            pose for the given camera.
+        """
+        if index < 0 or index >= len(self):
+            raise IndexError(f"Image index {index} is invalid")
+
+        wTi = self._wTi[index]
+        return wTi
+
+    def is_valid_pair(self, idx1: int, idx2: int) -> bool:
+        """Checks if (idx1, idx2) is a valid pair. idx1 < idx2 is required.
+
+        Args:
+            idx1: first index of the pair.
```

Comment:
Small nit: Google style would say first word should be capitalized, throughout this file :-)

“idx1: First index of the pair.”

Reply:
copilot should get better at this :(

Reply:
Done